pkgs/applications/science/molecular-dynamics/lammps/default.nix

```diff
@@ -0,0 +1,52 @@
+{ stdenv, writeText, fetchurl,
+  libpng, fftw,
+  mpiSupport ? false, mpi ? null
+}:
+
+assert mpiSupport -> mpi != null;
+
+stdenv.mkDerivation rec {
+  version = "16Feb16";
+  name = "lammps-${version}";
+
+  src = fetchurl {
+    url = "http://downloads.sourceforge.net/project/lammps/${ name }.tar.gz";
+    sha256 = "058459f3053cca0b8c0fdd05cca7acab6ac00934c6da8c9008aa0f55fb5ceefb";
+  };
+
+  passthru = {
+    inherit mpi;
+  };
+
+  buildInputs = [ fftw libpng ]
+  ++ (stdenv.lib.optionals mpiSupport [ mpi ]);
+
+  builder = writeText "builder.sh" ''
```

Comment:
The last time I asked someone whether it is possible to use standard phase hooks, it ended badly, but still: is there a particular reason for not using standard phase hooks here?